src/app/games/tafl/TaflRules.ts

```diff
@@ -0,0 +1,462 @@
+        const LOCAL_VERBOSE: boolean = false;
+        const listMoves: TaflMove[] = [];
+        const listPawns: Coord[] = this.getPlayerListPawns(player, state.board);
+        display(TaflRules.VERBOSE || LOCAL_VERBOSE, { TablutRules_getPlayerListMoves: { player, state } });
+
+        let pawnDestinations: Coord[];
+        let newMove: TaflMove;
+        for (const pawn of listPawns) {
+            pawnDestinations = this.getPossibleDestinations(pawn, state);
+            for (const destination of pawnDestinations) {
+                newMove = this.moveGenerator(pawn, destination);
+                listMoves.push(newMove);
+            }
+        }
+        return listMoves;
+    }
+    public getPlayerListPawns(player: Player, board: Table<TaflPawn>): Coord[] {
+        const listPawn: Coord[] = [];
+        let pawn: Coord;
+        let owner: Player;
+        for (let y: number = 0; y < this.config.WIDTH; y++) {
+            for (let x: number = 0; x < this.config.WIDTH; x++) {
+                // pour chaque case
+                pawn = new Coord(x, y);
+                owner = this.getAbsoluteOwner(pawn, board);
+                if (owner === player) {
+                    listPawn.push(pawn);
+                }
+            }
+        }
+        return listPawn;
+    }
+    public getPossibleDestinations(depart: Coord, state: S): Coord[] {
+        // search the possible destinations for the pawn at "depart"
+        const destinations: Coord[] = [];
+        let foundDestination: Coord;
+        for (const dir of Orthogonal.ORTHOGONALS) {
+            // we look for empty existing destinations in each direction as far as we can
+            foundDestination = depart.getNext(dir, 1);
+            let obstacleFound: boolean = false;
+            while (foundDestination.isInRange(this.config.WIDTH, this.config.WIDTH) &&
+                   obstacleFound === false)
+            {
+                const destinationEmpty: boolean = this.getAbsoluteOwner(foundDestination, state.board) === Player.NONE;
+                if (destinationEmpty) {
+                    if (this.isExternalThrone(foundDestination)) {
+                        if (state.board[depart.y][depart.x].isKing()) {
+                            destinations.push(foundDestination);
+                        }
+                    } else if (state.isCentralThrone(foundDestination)) {
+                        if (state.board[depart.y][depart.x].isKing() &&
+                            this.config.CASTLE_IS_LEFT_FOR_GOOD === false)
+                        {
+                            destinations.push(foundDestination);
+                        }
+                    } else {
+                        destinations.push(foundDestination);
+                    }
+                } else {
+                    obstacleFound = true;
+                }
+                foundDestination = foundDestination.getNext(dir, 1);
```

Comment:
useless assignment? foundDestination will be recomputed at the beginning of the loop

Reply:
C'est assigné dans le for, à l'intérieur y a un while qui déplace foundDestination de un pas dans la direction dir